### PR TITLE
Enhancement/ocr

### PR DESCRIPTION
**Pull request by ayyubibrahimi**
@pckhoi I've added an `ocr` and `ner` stage that is not integrated into deba, yet. Still working on the how. That said, before I went any further, I wanted to show you the following to confirm that integration is neccesary. I ask because I don't imagine having to ocr new documents more than twice a year, during which time ocr and ner may improve.

### CODE REVIEW

**Comment by pckhoi** [comment]
Even something that run twice a year should be version-controlled somewhere. If you're just keeping the scripts on your local computer, well what if it becomes broken and you can no longer access your work?

Regarding OCR, let me open a pull request of my own that has pytesseract setup in an OS-independent way. It also has an OCR cache that is DVC-controlled so if someone else already OCRed something, you don't need to run that step again on your local computer.

**Comment by pckhoi** [comment]
Please look at #312. Following the workflow laid out in that PR has several benefits:
- Metadata in a uniform format are generated for raw files.
- Painless OCR with pre-written library modules
- Shareable OCR cache
- Nice and clean progress bar during OCR

If you are fine with that workflow, please approve the PR, then rewrite the OCR script to follow this workflow. I suggest starting from scratch and copy over functions that you want to preserve.

**Comment by ayyubibrahimi** [comment]
@pckhoi Ready for review,

File: lib/columns.py [hunk 1]
```diff
@@ -320,3 +320,27 @@ def rearrange_property_claims_columns(df):
         "property_claims",
         df.sort_values(["agency", "property_claims_uid"]),
     )
+
+
+
+
+def rearrange_post_columns(df):
+    """Performs final processing step for an appeal hearing table
```
Comment (pckhoi) [inline lib/columns.py:328]: This function is not used anywhere.

File: dvc/data/raw/ner.dvc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+wdir: ../../../data/raw
+outs:
+- md5: 43d5c9ca303e4eccba73044b10eef279.dir
+  size: 262617
+  nfiles: 1
+  path: ner
```
Comment (pckhoi) [inline dvc/data/raw/ner.dvc:6]: Doesn't look like any script is reading files from `raw/ner`. What is the purpose of this file?

Comment (ayyubibrahimi) [reply dvc/data/raw/ner.dvc]: Old files from when I was experimenting prior to you setting up the work flow. Deleted.

File: deba.yaml [hunk 3]
```diff
@@ -27,7 +28,11 @@ targets:
...
       - meta/minutes_files.csv
+      - meta/post_officer_history_reports_files.csv
     prerequisites:
       - $(DEBA_MD5_DIR)/meta/minutes.py.md5
+      - $(DEBA_MD5_DIR)/meta/post_officer_history_reports.py.md5
       - $(DEBA_MD5_DIR)/raw_minutes.dvc.md5
-    recipe: "$(call deba_execute,meta/minutes.py)"
+      - $(DEBA_MD5_DIR)/raw_post_officer_history_reports.dvc.md5
+    recipe: "$(call deba_execute,meta/minutes.py)" 
+    recipe: "$(call deba_execute,meta/post_officer_history_reports.py)"
```
[the remark this answers is not in the document]
Comment (ayyubibrahimi) [reply deba.yaml:40]: Thanks!

File: datavalid.yml [hunk 4]
```diff
@@ -488,5 +490,74 @@ schemas:
...
+      - name: uid_8
+      - name: matched_uid
+      - name: matched_uid_1
+      - name: matched_uid_2
+      - name: matched_uid_3
+      - name: matched_uid_4
+      - name: matched_uid_5
+      - name: matched_uid_6
+      - name: matched_uid_7
+      - name: matched_uid_8
```
Comment (pckhoi) [inline datavalid.yml:561]: The changes in this file seem like they come from other branches. Sorry I should have detected it earlier. Can you please remove the changes in this file.

File: ner/post_officer_history_reports.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                    if key+str(num+1) not in tuples:
+                        tuples.remove(key) 
+                        tuples.insert(idx+num, key+ "_" + str(num+1))
+        
+        renamed_ents = dict(zip(tuples,[i[1] for i in ents]))
+        entities.append(renamed_ents)
+    
+    ner = pd.DataFrame(entities)
+    pdfs = pdfs.join(ner)
+    return ner, pdfs
```
Comment (pckhoi) [inline ner/post_officer_history_reports.py:79]: Joining pdfs to ner and output to `ner/post_officer_history_reports_pdfs.csv` seems redundant to me. Couldn't you just join them in any script that need the joined result?

**Comment by pckhoi** [comment]
Everything looks great! I trust that `make data/ner/post_officer_history_reports.csv` is working fine?

**Comment by ayyubibrahimi** [comment]
Yes, it's working as expected.

File: ner/post_officer_history_reports.py (hunk 5 above)
Comment (ayyubibrahimi) [reply ner/post_officer_history_reports.py]: Yes. Just addressed the redundancy. Thanks.

**Review by pckhoi — APPROVED** [review]
LGTM

**Pull request merged**